.github/workflows/main-checks.yml

```diff
@@ -146,6 +146,38 @@ jobs:
           RUSTFLAGS: ${{ matrix.toolchain == 'nightly' && '--cfg nightly_yew' || '' }}
         run: cargo test -p yew --all-features
 
+  example-runnable-tests-on-wasi:
+    name: Example Runnable Tests on WASI
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain:
+          - 1.67.0
+          - stable
+          - nightly
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          target: wasm32-wasi
+
+      - name: Install wasmtime
+        run: |
+          wget https://github.com/bytecodealliance/wasmtime/releases/download/v15.0.1/wasmtime-v15.0.1-x86_64-linux.tar.xz
+          tar xf wasmtime-v15.0.1-x86_64-linux.tar.xz
+          mv wasmtime-v15.0.1-x86_64-linux/wasmtime ~/wasmtime
+          rm -rf wasmtime-v15.0.1-x86_64-linux.tar.xz wasmtime-v15.0.1-x86_64-linux
+          chmod +x ~/wasmtime
+
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Build and run example wasi_ssr_module
+        run: |
+          cargo build --package wasi_ssr_module --target wasm32-wasi
+          ~/wasmtime --trap-unknown-imports target/wasm32-wasi/debug/wasi_ssr_module.wasm
```

Comment:
> We probably should prefer [`cargo-wasi`](https://github.com/bytecodealliance/cargo-wasi) instead of manually invoking wasmtime

> Commenting here because GitHub won't let me select the lines above this: can we run the native/non-browser tests in WASI as well?

In short, I currently think that it is not advisable to use `cargo wasi` instead of manually downloading the `wasmtime` CLI.

I've tried it many times in my own environment. Whether under Windows or Linux systems, `cargo wasi run` command cannot correctly find the compiled target file when encountering the specified package (`cargo wasi run -p wasi_ssr_module`). Although it prints the correct file path when reporting an error...

When I was looking for the answer to this question, I found that `cargo wasi run` actually [requires the `wasmtime` CLI to be installed locally before running](https://github.com/bytecodealliance/cargo-wasi/blob/e811d4889b516799d2f97813d073cec68b862adf/src/lib.rs#L150). In other words, instead of installing additional `cargo-wasi`, it is better to use `wasmtime` directly.